### PR TITLE
[SELC - 3922] feat: moved recipient code check and added check if institution is pr…

### PR DESCRIPTION
**Pull request by gianmarcoplutino**
…esent on IPA

#### List of Changes

- Moved recipientCode check in customValidate method
- added check if institution is present on IPA before create it
- implemented unit test

#### Motivation and Context

These changes were necessary as they had been made respectively in the Core microservice and in the onboarding-backend microservice but had not yet been transcribed into the new onboarding microservice

#### How Has This Been Tested?

the microservice was started locally, along with the onboarding-backend microservice, and onboarding calls were made to verify that the controls are working

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/functions/onboarding-functions/env/dev/terraform.tfvars
Comment (manuraf) [inline infra/functions/onboarding-functions/env/dev/terraform.tfvars:1]: we should also add for uat and prod

**Review by manuraf — CHANGES_REQUESTED** [review]

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 1]
```diff
@@ -129,9 +138,19 @@ private InstitutionResponse createInstitution(Institution institution, String pr
         return institutionApi.createInstitutionUsingPOST1(institutionMapper.toInstitutionRequest(institution));
     }
 
-    private boolean isGspAndProdInterop(InstitutionType institutionType, String productId) {
-        return InstitutionType.GSP == institutionType
-                && productId.equals(PROD_INTEROP.getValue());
+    private boolean isInstitutionPresentOnIpa(Institution institution) {
+        try {
+            if (institution.getSubunitType() != null && institution.getSubunitType() == InstitutionPaSubunitType.AOO) {
```
[the remark this answers is not in the document]
Comment (gianmarcoplutino) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java]: fix it

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 2]
```diff
@@ -129,9 +138,19 @@ private InstitutionResponse createInstitution(Institution institution, String pr
...
+                aooApi.findByUnicodeUsingGET(institution.getSubunitCode(), null);
+            } else if (institution.getSubunitType() != null && institution.getSubunitType() == InstitutionPaSubunitType.UO) {
```
[the remark this answers is not in the document]
Comment (gianmarcoplutino) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java]: fix it

File: infra/functions/onboarding-functions/env/dev/terraform.tfvars
Comment (gianmarcoplutino) [reply infra/functions/onboarding-functions/env/dev/terraform.tfvars:1]: i also added vars for uat and prod

**Review by manuraf — APPROVED** [review]

**Pull request merged**